lib/linguist/heuristics.yml

```diff
@@ -157,6 +157,19 @@ disambiguations:
     pattern: '# Microsoft Developer Studio Generated Build File'
   - language: Faust
     pattern: '\bprocess\s*[(=]|\b(library|import)\s*\(\s*"|\bdeclare\s+(name|version|author|copyright|license)\s+"'
+- extensions: ['.e']
+  rules:
+  - language: E
+    pattern:
+    - '^\s*(def|var)\s+(.+):='
+    - '^\s*(def|to)\s+(\w+)(\(.+\))?\s+{'
+    - '^\s*(when)\s+(\(.+\))\s+->\s+{'
+  - language: Eiffel
+    pattern:
+    - '^\s*{/\w+'
+    - '^\s*(across|elseif|feature|from|inherit|inspect|note|when)\s+'
```

Comment:
This pattern can be simplified (see my explanations below):

~~~suggestion
    - '^\s*(?:across|elseif|feature|from|inherit|inspect|note|when)\s'
~~~